### DISCUSSION TITLE
Where is file  /libraries/fof40/Utils/helpers.php ?

### DISCUSSION THREAD

**Discussion by korenevskiy** [joomla/joomla-cms#37664] [Q&A] [answered]
@nikosdion 
in this current Repository and in is [https://github.com/joomla-framework](https://github.com/joomla-framework)
I can't find this `/libraries/fof40/Utils/helpers.php` file anywhere.

**Comment by nikosdion** [comment, marked as answer]
This file is not part of Joomla itself, it belongs to FOF 4.0 which is a product of my company, Akeeba Ltd. You can find its repository at https://github.com/akeeba/fof. The file you are looking for is https://github.com/akeeba/fof/blob/development/fof/Utils/helpers.php (note that the development branch is FOF 4.x, the default branch of the repository is the old 3.x branch).

I also need to warn you that FOF is now considered deprecated and will become End of Life on August 17th, 2023 when Joomla 3 itself becomes End of Life. We have moved the development of all of our Joomla 4 software to the native Joomla 4 MVC.

**Reply by korenevskiy** [reply]
But I took this file from Joomla 4.
I wanted to add a PR with a new function to this file so that the function would be available in Joomla 4.
```PHP
function isTrue($value){
	if($value === false || $value === null || $value == 0 || $value == '' || $value == 'false' || $value == '0'
		 || $value == 'disabled' || $value == 'off' || $value == 'no' || $value == 'none')
		return false;
	return true;
}
```
This is an approximate writing of the function. But the need for such a feature in Joomla is obvious.
This function is needed to analyze the attribute values for XML.
each attribute must be repeatedly rechecked for all possible values. Despite the fact that the variants of values for all attributes are the same. The code becomes dirty, and repeated checks create a multiple little cognitive strain.
You constantly need to think about converting types to values when comparing, Which is very distracting from programming. 
And there is also no single rule for analyzing XML attributes. Even a young developer will now adapt to the analysis of this function.
Where do you think this feature can be added to?

**Reply by nikosdion** [reply]
I can tell you that /libraries/fof40 is our framework, FOF 4.0 (do you see where the name of the folder comes from now?). If you don't believe me, open the file and look at its header. It will tell you the same.

I wouldn't add functions to the root namespace. I would put this function as a static method in a helper. Off the top of my head, this _might_ fit the ApplicationHelper or the Joomla String package. You'd best ask some framework people, maybe @nibra would be able to guide you?

**Comment by korenevskiy** [comment]
@nibra Where do you think to place the function?
```PHP
function isTrue($value){
	if($value === false || $value === null || $value == 0 || $value == '' || $value == 'false' || $value == '0'
		 || $value == 'disabled' || $value == 'off' || $value == 'no' || $value == 'none')
		return false;
	return true;
}
```
This is an approximate writing of the function. But the need for such a feature in Joomla is obvious.
This function is needed to analyze the attribute values for XML.
each attribute must be repeatedly rechecked for all possible values. Despite the fact that the variants of values for all attributes are the same. The code becomes dirty, and repeated checks create a multiple little cognitive strain.
You constantly need to think about converting types to values when comparing, Which is very distracting from programming. 
And there is also no single rule for analyzing XML attributes. Even a young developer will now adapt to the analysis of this function.
Where do you think this feature can be added to?

**Reply by nibra** [reply]
Currently, the Framework does not have a place, where this functionality would belong, But I agree, it in general is useful; I use a sinilar construct in my projects to cast strings to boolean.

> You constantly need to think about converting types to values when comparing,

So maybe it is time to think about a package that solves this kind of issues. I started a dicussion for that here: [joomla-framework/discussions](https://github.com/orgs/joomla-framework/discussions/104).
